src/festim/hydrogen_transport_problem.py

```diff
@@ -369,6 +370,24 @@ def initialise_exports(self):
         a string, find species object in self.species"""
 
         for export in self.exports:
+            if isinstance(export, exports.VTXTemperatureExport):
+                if isinstance(self.temperature_fenics, (fem.Function, fem.Expression)):
+                    temperature_field = self.temperature_fenics
+                else:
+                    mesh = dolfinx.mesh.create_unit_square(MPI.COMM_WORLD, 1, 1)
+                    V = dolfinx.fem.FunctionSpace(mesh, ("CG", 1))
```

Comment:
Why create a unit square mesh everytime??? what if it's a 1D simulation?

You can use the mesh of the simulation directly:

```suggestion
                    V = dolfinx.fem.FunctionSpace(self.mesh, ("CG", 1))
```